config/github/src/main/kotlin/GitHubConfigFileProvider.kt

```diff
@@ -224,13 +252,33 @@ class GitHubConfigFileProvider(
     private fun sendHttpRequest(
         path: String,
         contentType: String
-    ) = runBlocking {
+    ): HttpResponse = runBlocking {
+        sendHttpRequestWithRetry(path, contentType)
+    }
+
+    /**
+     * Send a request to the GitHub REST API as defined by [baseUrl] with the provided [path] with a retry mechanism in
+     * case the request fails due to exceeded rate limits.
+     */
+    private tailrec suspend fun sendHttpRequestWithRetry(path: String, contentType: String): HttpResponse {
         val requestUrl = "$baseUrl$path"
         logger.debug("GET '{}'", requestUrl)
 
-        httpClient.get(requestUrl) {
+        val response = httpClient.get(requestUrl) {
             header("Accept", contentType)
         }
+
+        val rateLimitReset = checkForRateLimitRetry(response)
+        return if (rateLimitReset != null) {
+            val resetAt = Instant.fromEpochSeconds(rateLimitReset)
+            val delay = resetAt - Clock.System.now()
+            logger.warn("Rate limit exceeded. Retrying in {} seconds.", delay)
```

Comment:
Shouldn't a limit be put in place to avoid waiting for too long ?

Reply:
A limit would not really help, the next request would then fail again. Note that the restrictions for this mechanism are documented in the commit message.